components/invitation_modal/invite_as.tsx

```diff
@@ -23,6 +40,75 @@ export type Props = {
+    const license = useSelector(getLicense);
+    const dispatch = useDispatch<DispatchFunc>();
+    const subscription = useSelector((state: GlobalState) => state.entities.cloud.subscription);
+    const isSystemAdmin = useSelector(isCurrentUserSystemAdmin);
+    const hasPriorTrial = useSelector(checkHadPriorTrial);
+
+    let extraGuestLegend = true;
+    let guestDisabledClass = '';
+    let badges = null;
+    let guestDisabled = null;
+
+    const isCloud = license?.Cloud === 'true';
+    const isCloudFreeTrial = subscription?.is_free_trial === 'true';
+
+    const isPaidSubscription = isCloud && license?.SkuShortName !== LicenseSkus.Starter && !isCloudFreeTrial;
+
+    const closeInviteModal = () => {
+        dispatch(closeModal(ModalIdentifiers.INVITATION));
+    };
+
+    const restrictedIndicator = (
+        <RestrictedIndicator
+            blocked={!isCloudFreeTrial}
+            titleAdminPreTrial={formatMessage({
+                id: 'invite_modal.restricted_invite_guest.pre_trial_title',
+                defaultMessage: 'Try inviting guests with a free trial',
+            })}
+            messageAdminPreTrial={formatMessage({
+                id: 'invite_modal.restricted_invite_guest.pre_trial_description',
+                defaultMessage: 'Collaborate with users outside of your organization while tightly controlling their access to channels and team members. Get the full experience of Enterprise when you start a free, {trialLength} day trial.',
+            },
+            {trialLength: FREEMIUM_TO_ENTERPRISE_TRIAL_LENGTH_DAYS},
+            )}
+            titleAdminPostTrial={formatMessage({
+                id: 'invite_modal.restricted_invite_guest.post_trial_title',
+                defaultMessage: 'Upgrade to invite guest',
+            })}
+            messageAdminPostTrial={formatMessage({
+                id: 'invite_modal.restricted_invite_guest.post_trial_description',
+                defaultMessage: 'Collaborate with users outside of your organization while tightly controlling their access to channels and team members. Upgrade to the Professional plan to create unlimited user groups.',
+            })}
+            ctaExtraContent={(
+                <span className='badge-text'>
+                    {isCloudFreeTrial ? formatMessage({id: 'cloud_free.professional_feature.professional', defaultMessage: 'Professional feature'}) : formatMessage({id: 'cloud_free.professional_feature.try_free', defaultMessage: 'Professional feature- try it out free'})}
+                </span>
+            )}
+            clickCallback={closeInviteModal}
+        />
+    );
+
+    // show the badge logic (the restricted indicator takes care of the look when it is trial or not)
+    if (isSystemAdmin && isCloud && !isPaidSubscription) {
+        guestDisabledClass = isCloudFreeTrial ? '' : 'disabled-legend';
+        badges = {
+            matchVal: InviteType.GUEST as string,
+            badgeContent: restrictedIndicator,
```

Comment:
Defining the indicator inline here would ensure it's only created when actually required, right?

Reply:
Agree! my only concern is that the code will be harder to read.